lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,393 @@
+        post[:language] = 'en'
+        post[:command] = command
+        merchant = {}
+        merchant[:apiLogin] = @options[:api_login]
+        merchant[:apiKey] = @options[:api_key]
+        post[:merchant] = merchant
+      end
+
+      def add_transaction_type(post, type)
+        transaction = {}
+        transaction[:type] = type
+        post[:transaction] = transaction
+      end
+
+      def add_order(post, options)
+        order = {}
+        order[:accountId] = @options[:account_id]
+        order[:referenceCode] = options[:order_id] || SecureRandom.hex(16)
+        order[:description] = options[:description] || 'unspecified'
+        order[:language] = 'en'
+        post[:transaction][:order] = order
+      end
+
+      def add_buyer(post, options)
+        buyer = {}
+        if dni_number = options[:buyer_dni_number] || options[:dni_number]
+          buyer[:dniNumber] = dni_number
+        end
+        buyer[:cnpj] = options[:cnpj] if options[:cnpj]
+        buyer[:emailAddress] = options[:shipping_email] || options[:email] || 'unspecified@example.com'
+        if address = options[:shipping_address] || options[:billing_address] || options[:address]
+          buyer[:fullName] = address[:name]
+          shipping_address = {}
+          shipping_address[:street1] = (address[:address1] || '') + ' ' + (address[:address2] || '')
+          shipping_address[:city] = address[:city]
+          shipping_address[:state] = address[:state]
+          shipping_address[:country] = address[:country]
+          shipping_address[:postalCode] = address[:zip]
+          shipping_address[:phone] = address[:phone]
+          buyer[:shippingAddress] = shipping_address
+        end
+        post[:transaction][:order][:buyer] = buyer
+      end
+
+      def add_invoice(post, money, options)
+        additional_values = {}
+        tx_value = {}
+        tx_value[:value] = amount(money)
+        tx_value[:currency] = options[:currency] || currency(money)
+        additional_values[:TX_VALUE] = tx_value
+        post[:transaction][:order][:additionalValues] = additional_values
+      end
+
+      def add_signature(post)
+        post[:transaction][:order][:signature] = signature_from(post)
+      end
+
+      def signature_from(post)
+        signature_string = @options[:api_key] + '~' +
+                            @options[:merchant_id] + '~' +
+                            post[:transaction][:order][:referenceCode] + '~' +
+                            post[:transaction][:order][:additionalValues][:TX_VALUE][:value] + '~' +
+                            post[:transaction][:order][:additionalValues][:TX_VALUE][:currency]
+        Digest::MD5.hexdigest(signature_string)
+      end
+
+      def add_payment_method(post, payment_method)
+        if payment_method.is_a?(String)
+          _, brand, cvv, token = split_authorization(payment_method)
+          credit_card = {}
+          credit_card[:securityCode] = cvv
```

Comment:
Guess like a bunch of operations, it's difficult to tell since the sandbox doesn't support store.  😭